### PR TITLE
Footnotes in mei annot text

### PR DESCRIPTION
**Pull request by nikobeer**
(cherry picked from commit 23b070d050c6e6ff13609d9c775f867208438905)

# Conflicts:
#	add/data/xslt/meiP2html.xsl

### CODE REVIEW

**Review by riedde — COMMENTED** [review]
Thank you @nikobeer for this feature. I just added the css class "sup" you used in L. 67 to the todo.css

**Review by riedde — APPROVED** [review]
I did a quick test. Works fine!

**Comment by bwbohl** [comment]
Great, thanks!

Concerning the `sup` CSS class, it would be better to instantly add it to the respective scss file from the beginning. Also, textVieContent uses `.superscript` for that purpose which is already being allied in the annotation content box.

File: resources/css/todo.css [hunk 1]
```diff
@@ -202,6 +202,11 @@ span.italic {
     font-style: italic;
 }
 
+span.sup {
+    font-size: smaller;
+    vertical-align: super;
+}
+
```
Comment (bwbohl) [inline resources/css/todo.css:209]:
```suggestion
```

Remove as superfluous.

File: add/data/xslt/meiP2html.xsl [hunk 2]
```diff
@@ -26,6 +60,12 @@
             <xsl:apply-templates select="* | text()"/>
         </p>
     </xsl:template>
+    <xsl:template match="mei:annot[@type = 'note' and @place = 'foot']">
+        <xsl:variable name="footnoteCount">
+            <xsl:value-of select="functx:index-of-deep-equal-node($footnotes/mei:annot, .)"></xsl:value-of>
+        </xsl:variable>
+        <span class="sup"><xsl:value-of select="$footnoteCount"/></span>
```
Comment (bwbohl) [inline add/data/xslt/meiP2html.xsl:67]:
```suggestion
        <span class="superscript"><xsl:value-of select="$footnoteCount"/></span>
```

use existing superscript CSS class.

**Review by bwbohl — CHANGES_REQUESTED** [review]

File: add/data/xslt/meiP2html.xsl [hunk 3]
```diff
@@ -15,8 +12,45 @@
...
+    <xsl:variable name="footnotesBlock">
+        <xsl:for-each select="$footnotes/mei:annot">
+            <xsl:variable name="counter">
+                <xsl:value-of select="position()"></xsl:value-of>
+            </xsl:variable>
+            <xsl:variable name="content">
+                <xsl:apply-templates/>
+            </xsl:variable>
+            <div style="display:flex; color: grey; font-size: smaller; margin-bottom:0.5rem;">
+                <div style="display: inline; vertical-align: super; font-size: smaller;">
```
Comment (bwbohl) [inline add/data/xslt/meiP2html.xsl:29]: I just realised the direct CSS formatting of the footnote number. Couldn't we use `.superscript` here, too? Or may you define a class `.footnote-index` or similar?

Comment (riedde) [reply add/data/xslt/meiP2html.xsl:29]: We'll take the `.superscript` class. (Decision at meeting)

**Review by riedde — DISMISSED** [review]
I added the superscript to sass so it can handle both numbering inside an annotView content

File: packages/eoTheme/sass/etc/annotation.scss [hunk 4]
```diff
@@ -80,6 +80,11 @@ $content_baseFont_size: 14px;
 		    font-size: 1.1em;
 		}
 		
+		.superscript{
```
Comment (bwbohl) [inline packages/eoTheme/sass/etc/annotation.scss:83]: superscript was already defined for textVieContent and consequently worked for annotations, too (at least for spans). This more general rule does not break anything but we should consider reworking the SASS files at some point.

File: add/data/xslt/meiP2html.xsl (hunk 3 above)
Comment (bwbohl) [reply add/data/xslt/meiP2html.xsl:29]:
```suggestion
                <span class="superscript">
```

be aware that the closing tag has to be changed, too!

File: add/data/xslt/meiP2html.xsl [hunk 5]
```diff
@@ -15,8 +12,45 @@
...
+                    <xsl:value-of select="$counter"></xsl:value-of>
+                </div>
```
Comment (bwbohl) [inline add/data/xslt/meiP2html.xsl:31]:
```suggestion
                </span>
```

File: add/data/xslt/meiP2html.xsl [hunk 6]
```diff
@@ -15,8 +12,45 @@
...
+                <div style="display: inline; margin-left: .25rem;">
```
Comment (bwbohl) [inline add/data/xslt/meiP2html.xsl:32]: Is there a need for this to be a div with `display: inline`?

File: packages/eoTheme/sass/etc/annotation.scss (hunk 4 above)
Comment (riedde) [reply packages/eoTheme/sass/etc/annotation.scss:83]: In this case it is nested to annotations. But yes, we should work on this again!

File: add/data/xslt/meiP2html.xsl (hunk 6 above)
Comment (riedde) [reply add/data/xslt/meiP2html.xsl:32]: I'm not sure. I didn't remove it because I thought that divs could have another behaviour than spans

Comment (bwbohl) [reply add/data/xslt/meiP2html.xsl:32]: ok, hopefully it works with the span above :-/

**Review by bwbohl — APPROVED** [review]

**Pull request merged**